duckduckgo/duckduckgo-autofill
# Store cc & identities forms

**Reviewer:** @shakyShane 
**Asana:** https://app.asana.com/0/0/1201749050559820/f

## Description
Adds storage capabilities for credit cards and identities forms. We require a minimum set of values for each data type in order to create a usable entry. Namely:

- For credentials:
  - Only the password is required
- For identities:
  - Full name (either First + Last, or single field, in this second case we would store it as first name)
  - At least partial address:
    - street
    - city
    - ~~country~~ I decided against requiring the country because getting the country code is less reliable and we risk getting too many failures. We can easily add it later.
- For credit cards:
  - Card number 
  - Security code

More info [here](https://app.asana.com/0/0/1201530813714191/f).

I haven't changed how the form submission detection works at all. We can follow up on this separately.

## Steps to test
Added a bunch of tests for reading the form values. They're not exhaustive but a good start.

You can't check the whole on the app because it needs native work to accept the new data type and react to it.
